tests/config/test_parameters.py

```
@@ -133,6 +136,131 @@ def test_from_params_none_disables_pii():
     assert SafeSynthesizerParameters.from_params(replace_pii=None).replace_pii is None
 
 
+def test_from_params_accepts_explicit_dotted_path():
+    config = SafeSynthesizerParameters.from_params(**{"generation.validation.group_by_fix_unordered_records": True})
+
+    assert config.generation.validation.group_by_fix_unordered_records is True
+
+
+def test_from_params_rejects_top_level_none_before_nested_override():
+    with pytest.raises(
+        ParameterError, match="Cannot assign nested parameter path 'generation.num_records'.*'generation'"
```

Ruff (0.15.18)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)